lib/vtls/openssl.c

```
@@ -2172,9 +2170,11 @@ static int asn1_object_dump(ASN1_OBJECT *a, char *buf, size_t len)
 {
   int i, ilen;
 
-  if((ilen = (int)len) < 0)
+  if(len > (size_t)INT_MAX)
```

Using only lower bits could hide real error.